lib/gitlab/markdown.rb

```diff
@@ -64,7 +65,12 @@ def gfm_with_options(text, options = {}, project = @project, html_options = {})
         current_user:    current_user,
         only_path:       options[:reference_only_path],
         project:         project,
-        reference_class: html_options[:class]
+        reference_class: html_options[:class],
+
+        # RelativeLinkFilter
+        ref: @ref,
+        requested_path: @path,
+        project_wiki: @project_wiki
```

Comment:
@tsigo I have bad feeling from depending on instance variables here. What do you think? Is it safe to assume that these variables are always present?